File: 3300. Minimum Element After Replacement With Digit Sum/Solution.py
class Solution(object):
    def minElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = float('inf')
        for i in range(len(nums)):
            res = min(res,self.digitSum(nums[i]))

        return res


    def digitSum(self,num):
        sum = 0
        while num:
            sum+=num%10
            num = num//10

        return sum
        

File: 3300. Minimum Element After Replacement With Digit Sum/Solution.java
class Solution {
    public int minElement(int[] nums) {
        int res = Integer.MAX_VALUE;
        for(int i=0;i<nums.length;i++){
            nums[i] = digitSum(nums[i]);
            if(nums[i]<res){
                res = nums[i];
            }
        }

        return res;
    }
    private int digitSum(int n){
        int sum = 0;
        while(n!=0){
            sum += n%10;
            n/=10;
        }

        return sum;
    }
}